src/assets/CloudinaryFile.ts

```diff
@@ -237,14 +255,73 @@ class CloudinaryFile {
   }
 
 
+  /**
+   * Gets the signature of the URL
+   * This function returns the signature only when sign() was called and there's no authToken set.
+   */
   getSignature(): string {
-    if (this.signature) {
+    if (!this.signURL) { return; } // was sign() called? if not, skip this function
+    if (this?.authToken?.acl) { return; } // Do we have an auth token config set with acl? if so, skip this function
+    if (this.explicitAuthToken) { return; } // Do we have an explicit authToken set? if so, skip this function
+
+    if (this.signURL && this.signature) {
       return `s--${this.signature}--`;
     } else {
       return '';
     }
   }
 
+  /**
+   * Gets the full query param for the authToken signature
+   * - For client-side, this function uses the explicitAuthToken
+   * - For server-side, this function calculates the authToken given the authToken config
+   */
+  getAuthTokenQuery(): string {
+    if (!this.signURL) { return; }
+
+    // Do we have an explicit authToken? if so, use it.
+    if (this.explicitAuthToken) {
+      return `__cld_token__=${this.explicitAuthToken}`;
+    }
+
+    // Do we have an authToken config? if so, use it.
+    if (this.authToken) {
+      // To be implemented, server-side calculation of the authToken.
```

Comment:
I would not leave NOP code, would raise Unimplemented exception or something. Like, to make sure people get expected behavior.

Reply:
What would be the alternative? 
implementing the authToken calculation is beyond the scope of this PR, should we block this feature until it's we implement the token calculation?